tests/test_telemetry.py

```diff
@@ -375,6 +375,77 @@ def test_app_dependencies_loaded(self):
             if not seen:
                 raise Exception(dependency + " not recieved in app-dependencies-loaded message")
 
+    def setup_app_dependency_loaded_not_sent_dependency_collection_disabled(self):
+        weblog.get("/enable_product")
```

Comment:
I do not understand either, this endpoint is not implemented at all, but it's currently working ?

Reply:
GET /enable_product
This endpoint should enable all product DD_PROFILING_ENABLED=true, DD_APPSEC_ENABLED=true, DD_DYNAMIC_INSTRUMENTATION_ENABLED=true. But it is not yet implemented yet for any tracer.

Reply:
xdef setup_app_product_change(self):
        weblog.get("/enable_product")

    @missing_feature(
        context.library in ("dotnet", "nodejs", "java", "python", "golang", "cpp", "php", "ruby"),
        reason="Weblog GET/enable_product and app-product-change event is not implemented yet.",
    )
    def test_app_product_change(self):
        """Test product change data when product is enabled"""

        telemetry_data = list(interfaces.library.get_telemetry_data())
        if len(telemetry_data) == 0:
            raise Exception("No telemetry data to validate on")

        app_product_change_event_found = False

Reply:
I dont know. It my local branch its showing something different i see missing feature for all tracer.
def setup_app_product_change(self):
        weblog.get("/enable_product")
'''
    @missing_feature(
        context.library in ("dotnet", "nodejs", "java", "python", "golang", "cpp", "php", "ruby"),
        reason="Weblog GET/enable_product and app-product-change event is not implemented yet.",
    )
    def test_app_product_change(self):
        """Test product change data when product is enabled"""

        telemetry_data = list(interfaces.library.get_telemetry_data())
        if len(telemetry_data) == 0:
            raise Exception("No telemetry data to validate on")

        app_product_change_event_found = False
'''

Reply:
oh, it's because the function is called `setup_app_dependency_loaded_not_sent_dependency_collection_disabled` But the function `test_app_dependency_loaded_not_sent_dependency_collection_disabled`  does not exists, so the first one will never been called.

I read too fast, and presumed that the following method was the one called. 

You must either remove `setup_app_dependency_loaded_not_sent_dependency_collection_disabled`, or add `test_app_dependency_loaded_not_sent_dependency_collection_disabled` if it's you inteded to do

Reply:
Yeah, its wrong. I have fixed that, we dont need setup.